driver-core/src/main/com/mongodb/operation/AsyncQueryBatchCursor.java

```diff
@@ -298,6 +302,7 @@ private void handleGetMoreQueryResult(final AsyncConnection connection, final Si
                     connectionSource.release();
                 }
             }
+            AsyncQueryBatchCursor.this.release();
```

Comment:
It's not enough to just release here.  We also have to finish the cleanup if the instance has been closed while the operation was in progress.

Reply:
We now only release resources if no operation is in progress.